internal/staging/util.go

```diff
@@ -54,7 +61,32 @@ func (c *controller) getTestConfiguration(queue *s2hv1.Queue) *s2hv1.ConfigTestR
 		return nil
 	}
 
-	return deployConfig.TestRunner
+	testRunner := deployConfig.TestRunner
+
+	// override testRunner
+	if testRunnerOverrider := queue.GetTestRunnerExtraParameter(); testRunnerOverrider != nil {
+		testRunnerOverrider.Override(testRunner)
+
+		inferBranch := testRunnerOverrider.PullRequestInferGitlabMRBranch
+		canInferBranch := inferBranch != nil && *inferBranch
+
+		canQueryGitlab := testRunner != nil &&
+			testRunner.Gitlab != nil &&
+			testRunner.Gitlab.ProjectID != "" &&
+			testRunner.Gitlab.PipelineTriggerToken != ""
+
+		// infer branch name from MR in case of PRQueue
+		if queue.IsPullRequestQueue() && canInferBranch && canQueryGitlab {
+			gl := gitlab.NewClient(c.gitlabBaseURL, testRunner.Gitlab.PipelineTriggerToken)
+			branch, err := gl.GetMRSourceBranch(testRunner.Gitlab.ProjectID, queue.Spec.PRNumber)
+			// silently ignore error (in case of error, don't override the branch)
+			if err == nil && branch != "" {
```

Comment:
should we just log the error? in case the Gitlab server is down or something go wrong so we can have the visibility from the logs

Reply:
There's already an error log in `GetMRSourceBranch`. idk, wdyt?